mozart++/mpp_core/event_emitter.hpp

```diff
@@ -332,4 +333,22 @@ namespace mpp {
 #else
     using event_emitter = event_emitter_fast;
 #endif
+
+    namespace event {
+        extern event_emitter core_event;
+    }
+
+    template <typename T, typename... ArgsT>
+    void throw_ex(ArgsT &&... args) {
+        static_assert(std::is_base_of<std::exception, T>::value,
+                      "Only std::exception and its derived classes can be thrown");
+        T exception{std::forward<ArgsT>(args)...};
+        MOZART_LOGCR(exception.what())
+        event::core_event.emit("throw_ex", exception);
+#ifdef MOZART_NOEXCEPT
+        std::terminate();
+#else
+        throw exception;
+#endif
+    }
```

Comment:
This part of code is about exception handling using `event_emitter`.

I suggest put this part to `mpp_core/exception.hpp` because they are more related.